front/lib/api/wrappers.ts

```diff
@@ -152,19 +157,80 @@ export function withPublicAPIAuthentication<T, U extends boolean>(
     keyAuth: U extends true ? Authenticator : null
   ) => Promise<void> | void,
   opts: {
+    isWorkspaceRoute?: boolean;
+    forceAuthWithToken?: boolean;
     isStreaming?: boolean;
     allowUserOutsideCurrentWorkspace?: U;
-  } = {}
+  } = { isWorkspaceRoute: true, forceAuthWithToken: false }
 ) {
-  const { allowUserOutsideCurrentWorkspace, isStreaming } = opts;
+  const {
+    allowUserOutsideCurrentWorkspace,
+    isStreaming,
+    isWorkspaceRoute,
+    forceAuthWithToken,
+  } = opts;
+
+  type PublicAPIAllowedAuthType = "api_key" | "auth_token";
 
   return withLogging(
     async (
       req: NextApiRequest,
       res: NextApiResponse<WithAPIErrorResponse<T>>
     ) => {
-      const { wId } = req.query;
-      if (typeof wId !== "string" || !wId) {
+      const wId = typeof req.query.wId === "string" ? req.query.wId : undefined;
+
+      if (isWorkspaceRoute && !wId) {
+        return apiError(req, res, {
+          status_code: 404,
+          api_error: {
+            type: "workspace_not_found",
+            message: "The workspace was not found.",
+          },
+        });
+      }
+
+      const bearerTokenRes = await getBearerToken(req);
+      if (bearerTokenRes.isErr()) {
+        return apiError(req, res, {
+          status_code: 401,
+          api_error: {
+            type: "not_authenticated",
+            message:
+              "The request does not have valid authentication credentials.",
+          },
+        });
+      }
+      const bearerToken = bearerTokenRes.value;
+      const authMethod: PublicAPIAllowedAuthType = bearerToken.startsWith("sk-")
```

Comment:
No magic variable like `sk-` please, can we have it in a constant in `KeyResource` 🙏